### DISCUSSION TITLE
How to define Giraffe (1,4)-leaper?

### DISCUSSION THREAD

**Discussion by ilotari** [fairy-stockfish/Fairy-Stockfish#547] [Variants]
How do I define a Giraffe (1,4)-leaper (https://en.wikipedia.org/wiki/Giraffe_(chess))?

**Comment by ianfab** [comment]
Currently only the standard Betza atoms are supported, also see https://github.com/ianfab/Fairy-Stockfish/blob/0700dc6ab9e16c1e51e9101afd97bfb5b30dec86/src/piece.cpp#L33-L44

For reference, if someone intends to add another atom to the code manually, please be careful to consider https://github.com/ianfab/Fairy-Stockfish/blob/0700dc6ab9e16c1e51e9101afd97bfb5b30dec86/src/bitboard.cpp#L196-L202 since this currently assumes jumps larger than distance 3 are to be ignored, and will need some thought how to keep this check safe with larger jumps.
